File: library/src/main/ts/libraryBrs/src/app/core/services/rest/reservations.service.ts
import { Injectable } from '@angular/core';
import { BehaviorSubject, Observable } from 'rxjs';
import { Reservation, Response } from './api/api';
import { RestService } from './rest.service';
const URL = "api/reservations";

@Injectable()
export class ReservationsService {
  public reservations: BehaviorSubject<Reservation[]> = new BehaviorSubject<Reservation[]>([]);

  constructor(private restService: RestService) { }

  create(reservation: Reservation): Observable<Response<any>> {
    return this.restService.post(`${URL}`, reservation);
  }

  getByUserId(userId: string): any {
    return this.restService.get<Response<Reservation[]>>(`${URL}/user/${userId}`).subscribe((resp: Response<Reservation[]>) => {
      if (resp.content) {
        this.reservations.next(resp.content);
      }
    })
  }

  cancel(id:string): Observable<Response<any>> {
    return this.restService.patch(`${URL}/${id}/cancel`, {});
  }
}

File: library/src/main/dart/mobile/reader/lib/services/reservationsService.dart
import 'dart:collection';

import 'package:flutter/cupertino.dart';
import 'package:reader/services/api/api.dart';
import 'package:reader/services/restService.dart';

class ReservationsService extends ChangeNotifier {
  final String _url = "api/reservations";

  final List<Reservation> _reservation = [];

  UnmodifiableListView<Reservation> get items =>
      UnmodifiableListView(_reservation);

  void getByUserId(String userId) {
    RestService rest = RestService();
    rest.get<dynamic>(
        path: "${_url}/user/${userId}", onSuccess: onSuccessGetByUserId);
  }

  void onSuccessGetByUserId(dynamic object) {
    print(object);
    Response response = Response.fromJson(object);
    notifyListeners();
  }
}
